src/compass/s1_geo_stack.py

```diff
@@ -0,0 +1,468 @@
+                        continue
+                else:
+                    # Initialize geogrid with the info checked at this stage
+                    geogrid = isce3.product.bbox_to_geogrid(
+                        burst.as_isce3_radargrid(),
+                        burst.orbit,
+                        isce3.core.LUT2d(),
+                        x_spac, -y_spac,
+                        epsg)
+                    bbox_utm = snap_geogrid(geogrid, x_spac, y_spac)
+
+                burst_map['burst_id'].append(burst.burst_id)
+                # keep the burst object so we don't have to re-parse
+                burst_map['burst'].append(burst)
+
+                left, bottom, right, top = bbox_utm
+                burst_map['x_top_left'].append(left)
+                burst_map['y_top_left'].append(top)
+                burst_map['x_bottom_right'].append(right)
+                burst_map['y_bottom_right'].append(bottom)
+
+                burst_map['epsg'].append(epsg)
+                burst_map['date'].append(burst.sensing_start.strftime("%Y%m%d"))
+                # Save the file paths for creating the runconfig
+                burst_map['orbit_path'].append(orbit_path)
+                burst_map['zip_file'].append(zip_file)
+
+    burst_map = pd.DataFrame(data=burst_map)
+    return burst_map
+
+
+def snap_geogrid(geogrid, x_spac, y_spac):
+    """Snap the bounding box of the geogrid to integer multiples of the
+    spacing in X-/Y-directions.
+
+    Parameters
+    ----------
+    geogrid : isce3.core.Geogrid
+        Geogrid object
+    x_spac : float
+        Spacing of geocoded burst along X-direction
+    y_spac : float
+        Spacing of geocoded burst along Y-direction
+
+    Returns
+    -------
+    tuple
+        Tuple containing the snapped bounding box coordinates
+        (left, bottom, right, top)
+    """
+    x_left = x_spac * np.floor(geogrid.start_x / x_spac)
+    x_right = x_spac * np.ceil((geogrid.start_x + x_spac * geogrid.width) / x_spac)
+    y_bottom = y_spac * np.floor((geogrid.start_y + y_spac * geogrid.length) / y_spac)
+    y_top = y_spac * np.ceil(geogrid.start_y / y_spac)
+    return (x_left, y_bottom, x_right, y_top)
+
+
+
+
+
+def prune_dataframe(data, id_col, id_list):
+    '''
+    Prune dataframe based on column ID and list of value
+    Parameters:
+    ----------
+    data: pandas.DataFrame
+        dataframe that needs to be pruned
+    id_col: str
+        column identification for 'data' (e.g. 'burst_id')
+    id_list: list
+        List of elements to consider when pruning.
+        If exclude_items is False (default), then all elements in `data`
+            will be kept *except for* those in `id_list`.
+        If exclude_items is True, the items in `id_list` will be removed from `data`.
+    exclude_items: bool
+        If True, the items in `id_list` will be removed from `data`.
+        If False, all elements in `data` will be kept *except for* those in `id_list`.
+    Returns:
```

Comment:
```suggestion

    Parameters:
    ----------
    data: pandas.DataFrame
        dataframe that needs to be pruned
    id_col: str
        column identification for 'data' (e.g. 'burst_id')
    id_list: list
        List of elements to consider when pruning.
        If exclude_items is False (default), then all elements in `data`
            will be kept *except for* those in `id_list`.
        If exclude_items is True, the items in `id_list` will be removed from `data`.
    exclude_items: bool
        If True, the items in `id_list` will be removed from `data`.
        If False, all elements in `data` will be kept *except for* those in `id_list`.
        
    Returns:
```
Empty lines for readability.